extensions/security/deployment/src/test/java/io/quarkus/security/test/permissionsallowed/CustomPermissionWithMultipleArgs.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.security.test.permissionsallowed;
+
+import java.security.BasicPermission;
+import java.security.Permission;
+
+public class CustomPermissionWithMultipleArgs extends BasicPermission {
+
+    public static final String EXPECTED_FIELD_STRING_ARGUMENT = "expectedFieldStringArgument";
+    public static final int EXPECTED_FIELD_INT_ARGUMENT = 100;
+    public static final long EXPECTED_FIELD_LONG_ARGUMENT = 357;
+
+    private final String arg;
+    private final int fourth;
+    private final long first;
+
+    public CustomPermissionWithMultipleArgs(String name, String propertyOne, int fourth, long first) {
+        super(name);
+        this.arg = propertyOne;
+        this.first = first;
+        this.fourth = fourth;
+    }
+
+    @Override
+    public boolean implies(Permission p) {
```

Comment:
Ah, this one ignores the parameter and only uses `this.` members. Interesting!

Reply:
yeah, but this is just a test so... :-)